### ISSUE TITLE
Possible bug in shapenet Textures/Lighting/Loading Materials

### ISSUE THREAD

**Issue by ck-amrahd** [facebookresearch/pytorch3d#731]
Hi guys,
I am using the following code to render the shapenet dataset using pytorch3d:

        SHAPENET_PATH = "/home/user/data/ShapeNetCore.v2"
        shapenet_dataset = ShapeNetCore(SHAPENET_PATH, synsets=["car"], version=2, load_textures=True, texture_resolution=8)
        shapenet_model = shapenet_dataset[11]
        model_id = shapenet_model["model_id"]
        R, T = look_at_view_transform(1.0, 0.0, 90)
        cameras = FoVPerspectiveCameras(R=R, T=T, device=device)
        raster_settings = RasterizationSettings(image_size=1024)
        lights = PointLights(location=torch.tensor([2.0, 2.0, 2.0], device=device)[None], device=str(device))
        rendered_image = shapenet_dataset.render(
            model_ids=[model_id],
            device=device,
            cameras=cameras,
            raster_settings=raster_settings,
            lights=lights,
        )
        
        plt.figure(figsize=(7, 7))
        plt.imshow(rendered_image[0, ..., :3].cpu().numpy())
        plt.title("Rendered with texture")
        plt.grid("off")
        plt.axis("off")
        plt.show()

It renders the image as follows: 
![image](https://user-images.githubusercontent.com/60194966/123524643-980bcc00-d680-11eb-91fc-f16cf58da782.png)

But if I use a blender to load the same object, I get the following:

![image](https://user-images.githubusercontent.com/60194966/123524682-d4d7c300-d680-11eb-9f86-a00da912ea6c.png)

which seems a lot better than the one rendered from pytorch. Am I missing something in my code?

**Comment by ck-amrahd** [comment]
Hi guys, I tested on multiple obj files and there's a clear problem with pytorch3d rendering on the shapenet dataset. I am not sure where the problem is either on texture loading, material loading, or how rendering is done. I tested with: 

    https://github.com/panmari/stanford-shapenet-renderer

as well.

**Comment by ck-amrahd** [comment]
Hi @nikhilaravi , any update on this? Thank you.

**Comment by ck-amrahd** [comment]
Well, it's embarrassing. could you guys at least respond?

**Comment by gkioxari** [comment]
We are looking into this and also follow the same issue in https://github.com/facebookresearch/pytorch3d/issues/666

**Comment by ck-amrahd** [comment]
Turns out it's not a problem in pytorch3d, but the shapenet models being messy. I tried on meshlab and if I select face shading instead of vert shading, the problem goes out. The messy shapenet models don't allow for smooth interpolation of face normals. I used hardflatshader and changed some lighting parameters, and it is solved now. Thank you.
